diesel_cli/src/cli.rs

```diff
@@ -18,11 +18,41 @@ pub fn build_cli() -> App<'static, 'static> {
         .setting(AppSettings::VersionlessSubcommands)
         .arg(migration_dir_arg())
         .subcommand(SubCommand::with_name("run").about("Runs all pending migrations"))
-        .subcommand(SubCommand::with_name("revert").about("Reverts the latest run migration"))
-        .subcommand(SubCommand::with_name("redo").about(
-            "Reverts and re-runs the latest migration. Useful \
-             for testing that a migration can in fact be reverted.",
-        ))
+        .subcommand(
+            SubCommand::with_name("revert")
```

Comment:
This subcommand should have another argument `all` which is exclusive with `number` arg. The `revert all` should revert all the migrations.

Reply:
Hello @pksunkara 

I tried to use `conflicts_with` and a `default_value`, but it doesn't work since the default value will "activate" the `Arg`. 

```rust
.subcommand(
            SubCommand::with_name("revert")
                .about("Reverts the specified migrations")
                .arg(
                    Arg::with_name("REVERT_ALL")
                        .long("all")
                        .short("a")
                        .help("Reverts previously run migration files.")
                        .takes_value(false)
                        .conflicts_with("REVERT_NUMBER"),
                )
                .arg(
                    Arg::with_name("REVERT_NUMBER")
                        .long("number")
                        .short("n")
                        .help("Reverts the last `n` migration files")
                        .long_help(
                            "When this option is specified the last `n` migration files \
                             will be reverted. By default revert the last one.",
                        )
                        .takes_value(true)
                        .default_value("1")
                        .validator(is_positive_int)
                        .conflicts_with("REVERT_ALL"),
                ),
```

This will throw an error : 

```
diesel migration revert --all
stderr: error: The argument '--all' cannot be used with '--number <REVERT_NUMBER>'
```

If you need it, I can push my current code with the failing test.

Reply:
That issue had been fixed but it's only available in 3.0 prereleases. https://github.com/clap-rs/clap/issues/1605

Reply:
Ok thank you for the info, so I suppose I will need to handle the default value by myself or the conflict.

Reply:
It's probably the best way to proceed and please add a comment saying that the logic should be removed when upgrading to  clap 3.0